src/app/activity/bounty/Claim.ts

```diff
@@ -68,56 +100,147 @@ const getDbHandler = async (request: ClaimRequest): Promise<{dbBountyResult: Bou
     }
 }
 
-const writeDbHandler = async (request: ClaimRequest, claimedByUser: GuildMember): Promise<void> => {
+const writeDbHandler = async (request: ClaimRequest, dbBountyResult: BountyCollection, claimedByUser: GuildMember): Promise<BountyCollection> => {
     const db: Db = await MongoDbUtils.connect('bountyboard');
-	const bountyCollection = db.collection('bounties');
+    const bountyCollection = db.collection('bounties');
+    let claimedBounty: BountyCollection;
+    const currentDate = (new Date()).toISOString();
+
+    // If claiming an evergreen bounty, create a copy and use that
+    if (dbBountyResult.evergreen) {
+        const bountyRec: BountyCollection = Object.assign({}, dbBountyResult);
+        bountyRec.parentId = bountyRec._id;
+        delete bountyRec._id;
+        delete bountyRec.isParent;
+        delete bountyRec.childrenIds;
+        delete bountyRec.claimLimit;
+        const claimedInsertResult = await bountyCollection.insertOne(bountyRec);
+        if (claimedInsertResult == null) {
+            Log.error('failed to create claimed bounty from evergreen');
+            throw new Error('Sorry something is not working, our devs are looking into it.');
+        }
+        claimedBounty = await bountyCollection.findOne({_id: claimedInsertResult.insertedId});
+        let updatedBountyResult: UpdateWriteOpResult = await bountyCollection.updateOne({ _id: new mongo.ObjectId(dbBountyResult._id) }, {
+            $push: {
+                childrenIds: claimedBounty._id
+            }
+        });
+        if (updatedBountyResult == null) {
+            Log.error('failed to update evergreen bounty with claimed Id');
+            throw new Error('Sorry something is not working, our devs are looking into it.');
+        }
 
-	const dbBountyResult: BountyCollection = await bountyCollection.findOne({
-		_id: new mongo.ObjectId(request.bountyId),
-		status: BountyStatus.open,
-	});
+        // Pull it back for second update
+        dbBountyResult = await bountyCollection.findOne({
+            _id: new mongo.ObjectId(dbBountyResult._id)
+        });
     
-    const currentDate = (new Date()).toISOString();
-	const writeResult: UpdateWriteOpResult = await bountyCollection.updateOne(dbBountyResult, {
-		$set: {
-			claimedBy: {
-				discordHandle: claimedByUser.user.tag,
-				discordId: claimedByUser.user.id,
-				iconUrl: claimedByUser.user.avatarURL(),
-			},
-			claimedAt: currentDate,
-			status: BountyStatus.in_progress,
-		},
-		$push: {
-			statusHistory: {
-				status: BountyStatus.in_progress,
-				setAt: currentDate,
-			},
-		},
-	});
+
+        // If we have hit the claim limit, close this bounty
+        if (dbBountyResult.claimLimit !== undefined) {
```

Comment:
I think you can put this logic before the second parent bounty get request (line 134), by checking for 
```
claimedCount >= dbBountyResult.claimLimit - 1
```

This approach, however, is probably more readable. What do you think?

Reply:
Done. I had it that way originally but moved it out when I ran into trouble with filters. Now that those are fixed, this is OK